spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/source/DataFrameWriteTestBase.java

```diff
@@ -98,7 +98,7 @@ private Dataset<Row> createDataset(List<GenericData.Record> records, Schema sche
     }
 
     // verify that the dataframe matches
-    assertThat(rows.size()).isEqualTo(records.size());
+    assertThat(rows).hasSize(records.size());
```

Comment:
```suggestion
    assertThat(rows).hasSameSizeAs(records);
```

the advantage here is that this will also show the content of actual/expected when the assertion ever fails

Reply:
can you also please update all the other places that are affected in this PR?

Reply:
Thanks for your suggestion. Addressed comments.